File: ai-hub-backend/src/models/projectModel.js
import mongoose from 'mongoose';

const projectSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true
  },
  status: {
    type: String,
    enum: ['active', 'inactive', 'archived'],
    default: 'active'
  },
  apiKey: {
    type: String,
    required: true,
  },
  ownerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Admin',
    required: true
  },
  adminAccess: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Admin'
  }]
}, { timestamps: true });

export default mongoose.model('Project', projectSchema);

File: backend/infra/db/mongoose/models/project_model.go
package models

import "time"

type ProjectDoc struct {
	ID          string    `bson:"_id"`
	Name        string    `bson:"name"`
	Status      string    `bson:"status"`
	APIKey      string    `bson:"apiKey"`
	OwnerId     string    `bson:"ownerId"`
	AdminAccess []string  `bson:"adminAccess"`
	CreatedAt   time.Time `bson:"createdAt"`
	UpdatedAt   time.Time `bson:"updatedAt"`
}
